### ISSUE TITLE
No Internet connection on Asuswrt-Merlin router

### ISSUE THREAD

**Issue by pmcarrion** [Control-D-Inc/ctrld#101]
Issue # 45 has returned in version 1.3.1: [Issue with ASUSWRT-Merlin router](https://github.com/Control-D-Inc/ctrld/issues/45)

I was unable to get an Internet connection on my devices until I uninstalled and installed ctrld on my router.

**Comment by yegors** [comment]
Please perform the troubleshooting guide and provide more actionable details when it's in a broken state. 

https://github.com/Control-D-Inc/ctrld/wiki/Troubleshooting-Guide

Without this data, we cannot know what's wrong on your specific device.

**Comment by pmcarrion** [comment]
My router lost Internet connection at 7:50am. For some reason, ctrld stopped working.

I did these tests while the issue was present.

```
root@RT-AX88U:/tmp/home/root# ps | grep ctrld
10426 root  666m S    /jffs/controld/ctrld run --cd $DID --iface=aut
18608 root  3424 S    grep ctrld
```

```
root@RT-AX88U:/tmp/home/root# ps w | grep ctrld
10426 root  666m S    /jffs/controld/ctrld run --cd $DID --iface=auto --homedir=/jffs/controld --config=/jffs/controld/ct
18733 root  3424 S    grep ctrld
```


```
root@RT-AX88U:/tmp/home/root# netstat -tupln | grep ctrld
tcp        0      0 :::5354                 :::*                    LISTEN      10426/ctrld
udp        0      0 0.0.0.0:5353            0.0.0.0:*                           10426/ctrld
udp        0      0 0.0.0.0:5353            0.0.0.0:*                           10426/ctrld
udp        0      0 0.0.0.0:5353            0.0.0.0:*                           10426/ctrld
udp        0      0 0.0.0.0:5353            0.0.0.0:*                           10426/ctrld
udp        0      0 0.0.0.0:5353            0.0.0.0:*                           10426/ctrld
udp        0      0 0.0.0.0:5353            0.0.0.0:*                           10426/ctrld
udp        0      0 0.0.0.0:5353            0.0.0.0:*                           10426/ctrld
udp        0      0 0.0.0.0:5353            0.0.0.0:*                           10426/ctrld
udp        0      0 0.0.0.0:5353            0.0.0.0:*                           10426/ctrld
udp        0      0 0.0.0.0:5353            0.0.0.0:*                           10426/ctrld
udp        0      0 :::5353                 :::*                                10426/ctrld
udp        0      0 :::5353                 :::*                                10426/ctrld
udp        0      0 :::5353                 :::*                                10426/ctrld
udp        0      0 :::5353                 :::*                                10426/ctrld
udp        0      0 :::5353                 :::*                                10426/ctrld
udp        0      0 :::5353                 :::*                                10426/ctrld
udp        0      0 :::5353                 :::*                                10426/ctrld
udp        0      0 :::5353                 :::*                                10426/ctrld
udp        0      0 :::5353                 :::*                                10426/ctrld
udp        0      0 :::5353                 :::*                                10426/ctrld
udp        0      0 :::5354                 :::*                                10426/ctrld
```


```
root@RT-AX88U:/tmp/home/root# nslookup verify.controld.com 127.0.0.1
Server:    127.0.0.1
Address 1: 127.0.0.1 localhost.localdomain

nslookup: can't resolve 'verify.controld.com'
```

```
root@RT-AX88U:/tmp/home/root# cat /etc/resolv.conf
nameserver 127.0.0.1
```


```
root@RT-AX88U:/tmp/home/root# netstat -tupln | grep ":53\b"
tcp        0      0 127.0.0.1:53            0.0.0.0:*               LISTEN      22416/dnsmasq
tcp        0      0 192.168.XXX.X:53         0.0.0.0:*               LISTEN      22416/dnsmasq
tcp        0      0 2001:1388:XXX:XXX::1:53 :::*                    LISTEN      22416/dnsmasq
tcp        0      0 ::1:53                  :::*                    LISTEN      22416/dnsmasq
tcp        0      0 fe80::7e10:c9ff:XXXX:XXXX:53 :::*                    LISTEN      22416/dnsmasq
udp        0      0 127.0.0.1:53            0.0.0.0:*                           22416/dnsmasq
udp        0      0 192.168.XXX.X:53         0.0.0.0:*                           22416/dnsmasq
udp        0      0 2001:1388:XXX:XXX::1:53 :::*                                22416/dnsmasq
udp        0      0 ::1:53                  :::*                                22416/dnsmasq
udp        0      0 fe80::7e10:c9ff:XXXX:XXXX:53 :::*                                22416/dnsmasq
```


```
root@RT-AX88U:/tmp/home/root# cat /jffs/scripts/dnsmasq.postconf
# GENERATED BY ctrld - DO NOT MODIFY

#!/bin/sh

config_file="$1"
. /usr/sbin/helper.sh

pid=$(cat /tmp/ctrld.pid 2>/dev/null)
if [ -n "$pid" ] && [ -f "/proc/${pid}/cmdline" ]; then
  pc_delete "servers-file" "$config_file"           # no WAN DNS settings
  pc_append "no-resolv" "$config_file"              # do not read /etc/resolv.conf
  # use ctrld as upstream
  pc_delete "server=" "$config_file"
  pc_append "server=127.0.0.1#5354" "$config_file"
  pc_delete "add-mac" "$config_file"
  pc_delete "add-subnet" "$config_file"
  pc_append "add-mac" "$config_file"                # add client mac
  pc_append "add-subnet=32,128" "$config_file"      # add client ip
  pc_delete "dnssec" "$config_file"                 # disable DNSSEC
  pc_delete "trust-anchor=" "$config_file"          # disable DNSSEC
	
  # For John fork
  pc_delete "resolv-file" "$config_file"            # no WAN DNS settings

  # Change /etc/resolv.conf, which may be changed by WAN DNS setup
  pc_delete "nameserver" /etc/resolv.conf
  pc_append "nameserver 127.0.0.1" /etc/resolv.conf

  exit 0
fi



# GENERATED BY ctrld - EOF
```


```
root@RT-AX88U:/tmp/home/root# nvram show | grep dnsmasq_
size: 81391 bytes (49681 left)
```


```
root@RT-AX88U:/tmp/home/root# cd /jffs/controld
root@RT-AX88U:/jffs/controld# ./ctrld status
Oct 29 17:15:29.000 NTC Service is running
```

Internet returned upon restarting ctrld.

```
root@RT-AX88U:/jffs/controld# ./ctrld restart
Oct 29 17:34:33.000 NTC Service restarted
```

**Comment by cheng2wei** [comment]
I'm having a similar issue on my Unifi Dream Machine Pro with v1.3.1. It seems like any temporary upstream outage will cause the ctrld process to permanently view all upstream endpoints as unreachable until process restart. Downgrading back to v1.3.0 mitigates this issue.

**Comment by pmcarrion** [comment]
@cheng2wei The issue has been fixed in the latest dev version.
Use this command to install it:
`sh -c 'sh -c "$(curl -sL https://api.controld.dev/dl)" -s $DID forced'`
Replace $DID with your resolver ID.

**Comment by cuonglm** [comment]
This is fixed in v1.3.2 release.
